### PR TITLE
issue_71: add documentation

### PR DESCRIPTION
**Pull request by artificiala**
According to issue #71,  docstrings documentation (for pythainlp **v2.0.5** (10 May 2019)) was added for the following modules:

1. pythainlp.tokenize
2. pythainlp.tag
3. pythainlp.word_vector
4. pythainlp.ulmfit
5. pythainlp.util
6. pythainlp.soundex
7. pythainlp.spell
8. pythainlp.transliterate
8. pythainlp.corpus
9. pythainlp.tools
10. pythainlp.summarize
11. pythainlp.spell

Code examples were added to most of the functions/methods (i.e. `tokenize.word_tokenize`, `tag.pos_tag`). Rerturn type for functions/methods were specified.  Also, brief explaination of the functionality for important functions/methods.

## Fixing issues:

- [x] fix formatting issues (pep8) (due 15 June)

## Todos:

__High Priority__ 

Due date Mon 10 June 2019

1. pythainlp.tokenize
- [x]  todo_tokenize_1: provide example for `sent_tokenize`
- [x]  todo_tokenize_2: provide example for `word_tokenize`
- [x]  todo_tokenize_3: provide example for `syllabus_tokenize`
- [x]  todo_tokenize_4: provide example for `subword_tokenize`
- [x]  todo_tokenize_5: briefly explain the algorithm of maximum matching (`newmm`)and cite the reference
- [x]  todo_tokenize_6: briefly explain the algorithm of `pyicu` and cite the reference
- [x]  todo_tokenize_7: briefly explain the algorithm of `etcc` and cite the reference
- [x]  todo_tokenize_8: briefly explain the algorithm of `deepcut` and cite the reference
- [x]  todo_tokenize_9: briefly explain the algorithm of `multicut` and cite the reference
- [x]  todo_tokenize_10: briefly explain the algorithm of longest matching (`longest`) and cite the reference
- [x]  todo_tokenize_11: Fix docstring format in tcc.py

2. pythainlp.tag
- [x] todo_tag_1: provide lists of NER, POS tags
- [x]  todo_tag_2: provide examples for  `pos_tag`
- [x]  todo_tag_3: provide examples for  `pos_tag_sents`
- [x]  todo_tag_4: provide examples for `pos_tag_provinces`
- [x]  todo_tag_5: formatting docstring examples for `named_entity.ThaiNameTagger`
- [x]  todo_tag_6: briefly explain `unigram` engine and cite the reference
- [x]  todo_tag_7: briefly explain `perceptron` engine and cite the reference
- [x]  todo_tag_8: briefly explain `artagger` engine and cite the reference
- [x]  todo_tag_9: briefly explain `orchid` and cite the reference
- [x]  todo_tag_10: briefly explain `orchid_ud` and cite the reference
- [x]  todo_tag_11: briefly explain `pud` and cite the reference
- [x]  todo_tag_12: briefly explain each tagger engines at the another section.
- [x]  todo_tag_13: add reference section

3. pythainlp.word_vector

- [x]  todo_word_vector_1: give example for `most_similar_cosmul`
- [x]  todo_word_vector_2: give example for `doesnt_match`
- [x]  todo_word_vector_3: give example for `similarity`
- [x]  todo_word_vector_4: give example for `sentence_vectorizer`

4. pythainlp.ulmfit

- [x]  todo_ulmfit_1: provide example for `document_vector`
-  todo_ulmfit_2: provide example for `merge_wgts` (Canceled)
- [x]  todo_ulmfit_3: explain and show example for `pythainlp.ulmfit.ThaiTokenizer`

5. pythainlp.util

- [x]  todo_word_util_1: provide example for `thaicheck`
- [x]  todo_word_util_2: provide example for `thaiword_to_num`
- [x]  todo_word_util_3: provide example for `thai_to_eng`
- [x]  todo_word_util_4: provide example for `rank`
- [x]  todo_word_util_5: provide example for `num_to_thaiword`
- [x]  todo_word_util_6: provide example for `now_reign_year`
- [x]  todo_word_util_7: provide example for `normalize`
- [x]  todo_word_util_8: provide example for `countthai`
- [x]  todo_word_util_9: provide example for `find_keyword`
- [x]  todo_word_util_10: provide example for  `bahttext`
- [x]  todo_word_util_11: provide example for  `collate` and briefly explain
- [x]  todo_word_util_12: rewrite docstrting for `isthaichar`
- [x]  todo_word_util_13: provide example for `arabic_digit_to_thai_digit`
- [x]  todo_word_util_14: provide example for `num_to_thaiword`
- [x]  todo_word_util_15: provide example for `deletetone`
- [x]  todo_word_util_16: provide example for `eng_to_thai`
- [x]  todo_word_util_17: provide example for `thai_digit_to_arabic_digit`
- [x]  todo_word_util_18: provide example for `reign_year_to_ad`
- [x]  todo_word_util_19: rewrite docstrting for `isthai`
- [x]  todo_word_util_20: provide example for  `text_to_arabic_digit`
- [x]  todo_word_util_21: provide example for `text_to_thai_digit`
- [x]  todo_word_util_22: provide example for `thai_strftime`
- [x]  todo_word_util_23: format docstring  for `thai_strftime`

-----------------------

__Medium Priority__

Due date: Thu 13 June 2019

6. pythainlp.soundex

- [x]  todo_soundex_1: provide more examples for `metasound`
- [x]  todo_soundex_2: provide examples for `udom83`
- [x]  todo_soundex_3: provide examples for `lk82`
- [x]  todo_soundex_4: provide examples for `soundex`
- [x]  todo_soundex_5: briefly explain `lk82`
- [x]  todo_soundex_6: briefly explain `udom82`
- [x]  todo_soundex_7: briefly explain `metasound`

7. pythainlp.spell

- [x]  todo_spell_1: briefly explain Peter Norvig’s algorithm
- [x]  todo_spell_2: cite the reference of  Peter Norvig’s algorithm
- [x]  todo_spell_3: provide examples for `correct`
- [x]  todo_spell_4: provide examples for `spell`
- [x]  todo_spell_5: provide examples for `NorvigSpellChecker.correct`
- [x]  todo_spell_6: provide examples for `NorvigSpellChecker.dictionary`
- [x]  todo_spell_7: provide examples for `NorvigSpellChecker.freq`
- [x]  todo_spell_8: provide examples for `NorvigSpellChecker.known`
- [x]  todo_spell_9: provide examples for `NorvigSpellChecker.prob`
- [x]  todo_spell_10: provide examples for `NorvigSpellChecker.spell`
- [x]  todo_spell_11:  briefly explain constant variable `DEFAULT_SPELL_CHECKER`
- [x]  todo_spell_12: explain `spell`

8. pythainlp.transliterate

- [x]  todo_transliterate_1: format docstring for `romanize`
- [x]  todo_transliterate_2: format docstring for `transliterate`
- [x]  todo_transliterate_3: provide examples for `romanize`
- [x]  todo_transliterate_4: provide examples for `transliterate`
- [x]  todo_transliterate_5: add reference

------------------------

__Low Priority__

Due date: Thu 13 June 2019

9. pythainlp.corpus

- [x]  todo_corpus_1: provide link to `thai_stopwords`
- [x]  todo_corpus_2: provide link to `thai_words`
- [x]  todo_corpus_3: provide link to `thai_syllables`
- [x]  todo_corpus_4: provide link to `thai_negations`
- [x]  todo_corpus_5: provide link to `countries`
- [x]  todo_corpus_6: provide link to `provinces`
- [x]  todo_corpus_7: provide example for `corpus.download`
- [x]  todo_corpus_8: provide example for `corpus.remove`
- [x]  todo_corpus_9: provide example for `corpus.get_corpus`
- [x]  todo_corpus_10: provide example for `corpus.get_corpus_path`
- [x]  todo_corpus_11: provide examples for `conceptnet.edges`
- [x]  todo_corpus_12: provide examples for `wordnet.synset`
- [x]  todo_corpus_13: provide examples for `wordnet.synsets`
- [x]  todo_corpus_14: provide examples for `wordnet.all_lemma_names`
- [x]  todo_corpus_15: provide examples for `wordnet.all_synsets`
- [x]  todo_corpus_16: provide examples for `wordnet.langs`
- [x]  todo_corpus_17: provide examples for `wordnet.lemmas`
- [x]  todo_corpus_18: provide examples for `wordnet.lemma`
- [x]  todo_corpus_19: provide examples for `wordnet.lemma_from_key`
- [x]  todo_corpus_20: provide examples for `wordnet.path_similarity`
- [x]  todo_corpus_21: provide examples for `wordnet.lch_similarity`
- [x]  todo_corpus_22: provide examples for `wordnet.wup_similarity`
- [x]  todo_corpus_23: provide examples for `wordnet.morphy`
- [x]  todo_corpus_24: briefly explain `conceptnet.edges`
- [x]  todo_corpus_25: briefly explain `wordnet.custom_lemmas`

10. pythainlp.tools

- [x]  provide example for `tools.get_full_data_path`
- [x]  provide example for `tools.get_pythainlp_data_path`
- [x]  provide examples for `tools.get_pythainlp_path`

11. pythainlp.summarize

- [x]  provide examples for `summarize.summarize`
- [x]  briefly explain functionality of `summarize.summarize`

### CODE REVIEW

**Comment by artificiala** [comment]
Todos:

- [ ] fix formatting issues (pep8)

File: docs/api/tokenize.rst [hunk 1]
```diff
@@ -16,14 +16,40 @@ Modules
 .. autoclass:: Tokenizer
    :members:
 
-NEWMM
------
+
+Tokenize Engine
```
Comment (heytitle) [inline docs/api/tokenize.rst]: As we have several engines, we should use `Engines` instead. Moreover, we should use a proper noun as `ADJ` for `Engines`. So, in short, I think the header should be `Tokenization Engines`.

**Comment by heytitle** [comment]
Hi,

I have a question regarding writing consistency. Currently, we tend to use both `segmentation` and `tokenization`. Although these words are interchangeable, do you think it's better if we're strict with one of them, i.e. `tokenization`?

File: docs/api/tokenize.rst (hunk 1 above)
Comment (artificiala) [reply docs/api/tokenize.rst]: Thanks for reviewing, I have changed from "Tokenize Engine" to "Tokenization Engines".

**Pull request merged**